File: android/src/main/kotlin/io/woodemi/quick_scan/QuickScanPlugin.kt
package io.woodemi.quick_scan

import io.flutter.plugin.common.PluginRegistry.Registrar

class QuickScanPlugin {
    companion object {
        @JvmStatic
        fun registerWith(registrar: Registrar) {
            val scanViewFactory = ScanViewFactory(registrar.messenger())
            registrar.platformViewRegistry().registerViewFactory("scan_view", scanViewFactory)
        }
    }
}

File: ios/Classes/QuickScanPlugin.m
#import "QuickScanPlugin.h"
#import "ScanViewFactory.h"

@implementation QuickScanPlugin
+ (void)registerWithRegistrar:(NSObject <FlutterPluginRegistrar> *)registrar {
    ScanViewFactory *scanViewFactory = [ScanViewFactory initWithMessenger:registrar.messenger];
    [registrar registerViewFactory:scanViewFactory withId:@"scan_view"];
}

@end
